homeassistant/components/monzo/coordinator.py

```diff
@@ -45,5 +46,10 @@ async def _async_update_data(self) -> MonzoData:
             pots = await self.api.user_account.pots()
         except AuthorisationExpiredError as err:
             raise ConfigEntryAuthFailed from err
+        except InvalidMonzoAPIResponseError as err:
+            message = "Invalid Monzo API response."
+            if err.missing_key:
+                message += f"\nMissing key: {err.missing_key} Response:\n{pformat(err.response)}"
+            raise HomeAssistantError(message) from err
```

Comment:
You should raise `UpdateFailed` here

Reply:
I did think that, but when using UpdateFailed we don't seem to get a message in the logs, only this:
![image](https://github.com/user-attachments/assets/815ee442-2835-45ce-a8c5-4a0e06358d2f)

which cuts off quite a bit of data I'd need to troubleshoot. Is there anywhere else the UpdateFailed message can be seen in full by the user? Or a way to get it to appear in the logs?

Reply:
Isn't this already logged?

Reply:
Not that I can see. Enabling debug logging for the integration will give:
![image](https://github.com/user-attachments/assets/960ab11f-6cdd-48b5-95f0-bde0cb4bd8ba)
but that's not really much help

Reply:
Note: "missing key: typadsae" looks like a typo in the library...
See my comment in https://github.com/home-assistant/core/pull/129726

Reply:
Resolved by using UpdateFailed, but also debug logging the full error